### ISSUE TITLE
secret and subpath may have same name in vault, how to handle?

### ISSUE THREAD

**Issue by muryoutaisuu** [Muryoutaisuu/secretsfs#25]
```
[bea@ ~]$ ls -la /mnt/secretsfs/secretsfiles/applam/gygerj
total 0
dr-xr-x--- 1 root root 0 Jan  1  1970 .
dr-xr-x--- 0 root root 0 Jan  1  1970 ..
dr-xr-x--- 0 root root 0 Jan  1  1970 E1
dr-xr-x--- 0 root root 0 Jan  1  1970 E1
[bea@ ~]$ ls -la /mnt/secretsfs/secretsfiles/applam/gygerj/E1
total 0
dr-xr-x--- 1 root root 0 Jan  1  1970 .
dr-xr-x--- 0 root root 0 Jan  1  1970 ..
dr-xr-x--- 0 root root 0 Jan  1  1970 sub-secret
[bea@ ~]$ ls -la /mnt/secretsfs/secretsfiles/applam/gygerj/E1/sub-secret/
ls: reading directory /mnt/secretsfs/secretsfiles/applam/gygerj/E1/sub-secret/: Input/output error
total 0
[bea@ ~]$ ls -la /mnt/secretsfs/secretsfiles/applam/gygerj/E1/sub-secret/
ls: reading directory /mnt/secretsfs/secretsfiles/applam/gygerj/E1/sub-secret/: Input/output error
total 0

```

**Comment by muryoutaisuu** [comment]
similar to #8, but not the same!

**Comment by muryoutaisuu** [comment]
still happening:

```
$ ls -la /mnt/secretsfs/secretsfiles/applam/gygerj
total 0
dr-xr-x--- 1 root root 0 Jan  1  1970 .
dr-xr-x--- 0 root root 0 Jan  1  1970 ..
dr-xr-x--- 0 root root 0 Jan  1  1970 E1
dr-xr-x--- 0 root root 0 Jan  1  1970 E1
dr-xr-x--- 0 root root 0 Jan  1  1970 fiorettin
```

**Comment by muryoutaisuu** [comment]
fixed in #31
